NETDefault/Gateway.Buckaroo/Mappers/BuckarooResponseMapper.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Gateway.Buckaroo.Models.Dto.Responses;
+using Gateway.Buckaroo.Models.Entities;
+
+namespace Gateway.Buckaroo.Mappers;
+
+internal static class BuckarooResponseMapper
+{
+    internal static Debtor Map(DebtorInfoResponseDto response)
+    {
+        var id = response?.Services?.FirstOrDefault(s => s.Name == "CreditManagement3")?.Parameters?.FirstOrDefault(p => p.Name == "Code")?.Value;
+        var guid = string.IsNullOrWhiteSpace(id)
+            ? Guid.Empty
+            : Guid.Parse(id);
+
+        return new Debtor
+        {
+            Id = guid,
+            FirstName = response?.Services?.FirstOrDefault(s => s.Name == "CreditManagement3")?.Parameters?.FirstOrDefault(p => p.Name == "FirstName")?.Value ?? "",

```

Comment:
There is a lot of repeating code here, and with those `FirstOrDefault` is a bit wasteful performance wise. I would opt to use a local variable here and store the CreditManagment parameter object somewhere.

Reply:
Good one!